src/net/ossl.cc

```diff
@@ -1451,9 +1483,26 @@ class session : public enable_shared_from_this<session>, public session_impl {
                 break;
             }
 
-            SSL_CTX_set_options(
-              ssl_ctx.get(), SSL_OP_ALL | SSL_OP_ALLOW_CLIENT_RENEGOTIATION);
+            auto options = SSL_OP_ALL | SSL_OP_ALLOW_CLIENT_RENEGOTIATION;
+            if (_creds->is_server_precedence_enabled()) {
+                options |= SSL_OP_CIPHER_SERVER_PREFERENCE;
+            }
+
+            SSL_CTX_set_options(ssl_ctx.get(), options);
+        } else {
+            if (_creds->is_server_precedence_enabled()) {
+                SSL_CTX_set_options(ssl_ctx.get(), SSL_OP_CIPHER_SERVER_PREFERENCE);
+            }
+        }
+
+        if (!SSL_CTX_set_min_proto_version(ssl_ctx.get(), TLS1_VERSION)) {
+            throw ossl_error::make_ossl_error(
+                "Failed to set minimum TLS version to 1"
+            );
         }
+        // This call is required to lower SSL's security level to permit TLSv1.0 and TLSv1.1
+        // See https://www.openssl.org/docs/man3.0/man3/SSL_CTX_set_security_level.html
+        SSL_CTX_set_security_level(ssl_ctx.get(), 0);
```

Comment:
Should this be configurable, or is an empty `ciphersuite` sufficient to disable TLS v1.2 and below?

Reply:
No, leaving either `ciphersuites` or `cipher_string` empty just results in a default.

There is another project for us to work on to allow the user to control which TLS versions are enabled.  Would probably cover that there.  This PR was just to get to feature parity with GnuTLS